test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -456,21 +461,32 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 			spec, err := json.Marshal(profile.Spec)
 			Expect(err).ToNot(HaveOccurred())
 
+			if profile.Spec.MachineConfigPoolSelector != nil {
+				By("Removing Machine Config Selector")
+				mcpSelectorRemoved = true
+				Expect(testclient.Client.Patch(context.TODO(), profile,
+					client.RawPatch(
+						types.JSONPatchType,
+						[]byte(fmt.Sprintf(`[{"op": "remove", "path": "/spec/%s"}]`, "machineConfigPoolSelector")),
+					),
+				)).ToNot(HaveOccurred())
+			}
+
 			By("Applying changes in performance profile and waiting until mcp will start updating")
 			Expect(testclient.Client.Patch(context.TODO(), profile,
 				client.RawPatch(
 					types.JSONPatchType,
 					[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, spec)),
 				),
 			)).ToNot(HaveOccurred())
+
 			mcps.WaitForCondition(newRole, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
 
 			By("Waiting when MCP finishes updates and verifying new node has updated configuration")
 			mcps.WaitForCondition(newRole, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
 
-			kblcfg, err := nodes.ExecCommandOnNode(chkKubeletConfig, newCnfNode)
-			Expect(err).ToNot(HaveOccurred(), "failed to execute %s", chkKubeletConfig)
-			Expect(kblcfg).To(ContainSubstring("topologyManagerPolicy"))
+			kubeletConfig, err := nodes.GetKubeletConfig(newCnfNode)
+			Expect(kubeletConfig.TopologyManagerPolicy).ToNot(BeEmpty())
```

Comment:
this seems like a good change and we can merge it separately anytime. But it will require the BZ dance so likely not worth the extra hassle